public/app/features/explore/TraceView/useDetailState.ts

```diff
@@ -18,12 +19,23 @@ export function useDetailState(frame: DataFrame) {
 
   const toggleDetail = useCallback(
     function toggleDetail(spanID: string) {
-      const newDetailStates = new Map(detailStates);
-      if (newDetailStates.has(spanID)) {
-        newDetailStates.delete(spanID);
+      let newDetailStates = new Map(detailStates);
+
+      if (config.featureToggles.traceViewDrawer) {
+        if (newDetailStates.has(spanID)) {
+          newDetailStates = new Map();
+        } else {
+          newDetailStates = new Map();
+          newDetailStates.set(spanID, new DetailState());
+        }
       } else {
-        newDetailStates.set(spanID, new DetailState());
+        if (newDetailStates.has(spanID)) {
+          newDetailStates.delete(spanID);
+        } else {
+          newDetailStates.set(spanID, new DetailState());
+        }
```

Comment:
Will essentially be removed (with other checks for config.featureToggles.traceViewDrawer being enabled) when toggle is removed.